src/discovery/mdns.js

```diff
@@ -0,0 +1,119 @@
+import { TypedEmitter } from 'tiny-typed-emitter'
+import net from 'node:net'
+import NoiseSecretStream from '@hyperswarm/secret-stream'
+import { once } from 'node:events'
+import dnssd from 'dnssd'
```

Comment:
This is no longer maintained. We have been using the fork [@gravitysoftware/dnssd](https://www.npmjs.com/package/@gravitysoftware/dnssd). Seth had written [some type declarations](https://github.com/digidem/multicast-service-discovery/blob/main/types/gravitysoftware__dnssd/index.d.ts).

There is also https://github.com/homebridge/ciao, but there seem to be some outstanding issues with that and maintenance seems to be focussed just on supporting the Homebridge app e.g. see https://github.com/SignalK/signalk-server/issues/1353.

There is also https://www.npmjs.com/package/bonjour-service but it doesn't seem to suggest any RFC compliance.

I think maybe @gravitysoftware/dnssd might be the most reliable option.